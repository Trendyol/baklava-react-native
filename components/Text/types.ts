export type TextVariantTypes =
  | 'heading1'
  | 'heading2'
  | 'heading3'
  | 'subtitle01Regular'
  | 'subtitle01Medium'
  | 'subtitle01Semibold'
  | 'subtitle01Bold'
  | 'subtitle02Regular'
  | 'subtitle02Medium'
  | 'subtitle02Semibold'
  | 'subtitle02Bold'
  | 'subtitle03Regular'
  | 'subtitle03Medium'
  | 'subtitle03Semibold'
  | 'subtitle03Bold'
  | 'subtitle04Regular'
  | 'subtitle04Medium'
  | 'subtitle04Semibold'
  | 'subtitle04Bold'
  | 'bodyText'
  | 'bodyUnderline'
  | 'bodyTextLink'
  | 'bodyLongText'
  | 'captionText'
  | 'captionMedium'
  | 'captionLongText'
  | 'captionTextLink';
