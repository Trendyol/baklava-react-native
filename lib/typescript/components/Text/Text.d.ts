declare function Text(props: any): JSX.Element;
declare namespace Text {
    var defaultProps: {
        variant: string;
        color: string;
    };
}
export default Text;
