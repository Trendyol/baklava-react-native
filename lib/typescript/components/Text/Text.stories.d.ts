import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from '../Text/Text';
declare const TextMeta: ComponentMeta<typeof Text>;
export default TextMeta;
declare type TextStory = ComponentStory<typeof Text>;
export declare const Heading: TextStory;
export declare const SubtitleXLarge: TextStory;
export declare const SubtitleLarge: TextStory;
export declare const SubtitleMedium: TextStory;
export declare const SubtitleSmall: TextStory;
export declare const Body: TextStory;
export declare const Caption: TextStory;
