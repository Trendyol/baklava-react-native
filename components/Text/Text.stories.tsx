import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from '../Text/Text';
import theme from '../../src/theme';

const variantList = Object.keys(theme.textStyle);

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radios' },
    },
  },
  args: {
    variant: variantList[0] as string,
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Basic: TextStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Text
    </Text>
    <Text variant={args.variant} p={3}>
      Sample Text
    </Text>
  </>
);

export const Heading: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Heading
    </Text>
    <Text variant="heading1" p={3}>
      Heading 1
    </Text>
    <Text variant="heading2" p={3}>
      Heading 2
    </Text>
    <Text variant="heading3" p={3}>
      Heading 3
    </Text>
  </>
);

export const Subtitle01: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle01Regular" p={3}>
      Subtitle 01 / Regular
    </Text>
    <Text variant="subtitle01Medium" p={3}>
      Subtitle 01 / Medium
    </Text>
    <Text variant="subtitle01Semibold" p={3}>
      Subtitle 01 / Semibold
    </Text>
    <Text variant="subtitle01Bold" p={3}>
      Subtitle 01 / Bold
    </Text>
  </>
);

export const Subtitle02: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle02Regular" p={3}>
      Subtitle 02 / Regular
    </Text>
    <Text variant="subtitle02Medium" p={3}>
      Subtitle 02 / Medium
    </Text>
    <Text variant="subtitle02Semibold" p={3}>
      Subtitle 02 / Semibold
    </Text>
    <Text variant="subtitle02Bold" p={3}>
      Sutitle 02 / Bold
    </Text>
  </>
);

export const Subtitle03: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle03Regular" p={3}>
      Subtitle 03 / Regular
    </Text>
    <Text variant="subtitle03Medium" p={3}>
      Subtitle 03 / Medium
    </Text>
    <Text variant="subtitle03Semibold" p={3}>
      Subtitle 03 / Semibold
    </Text>
    <Text variant="subtitle03Bold" p={3}>
      Subtitle 03 / Bold
    </Text>
  </>
);

export const Subtitle04: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle04Regular" p={3}>
      Subtitle 04 / Regular
    </Text>
    <Text variant="subtitle04Medium" p={3}>
      Subtitle 04 / Medium
    </Text>
    <Text variant="subtitle04Semibold" p={3}>
      Subtitle 04 / Semibold
    </Text>
    <Text variant="subtitle04Bold" p={3}>
      Subtitle 04 / Bold
    </Text>
  </>
);

export const Body: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Body
    </Text>
    <Text variant="bodyText" p={3}>
      Body / Text
    </Text>
    <Text variant="bodyUnderline" p={3}>
      Body / Underline
    </Text>
    <Text variant="bodyTextLink" p={3}>
      Body / Text Link
    </Text>
    <Text variant="bodyLongText" p={3}>
      Body / Long Text
    </Text>
  </>
);

export const Caption: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Caption
    </Text>
    <Text variant="captionText" p={3}>
      Caption / Text
    </Text>
    <Text variant="captionMedium" p={3}>
      Caption / Medium
    </Text>
    <Text variant="captionLongText" p={3}>
      Caption / Long Text
    </Text>
    <Text variant="captionTextLink" p={3}>
      Caption / Text Link
    </Text>
  </>
);
