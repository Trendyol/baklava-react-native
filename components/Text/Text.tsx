import React from 'react';
import { Text as T } from 'react-native';
import styled from 'styled-components/native';
import {
  compose,
  color as styledColor,
  space,
  typography,
  position,
  size,
  fontFamily,
  fontWeight,
  textStyle,
  variant as styledVariant,
  fontSize,
} from 'styled-system';
import theme from '../../src/theme';
import { TextVariantTypes } from './types';

const StyledText = styled(T)(
  compose(
    styledColor,
    space,
    typography,
    position,
    size,
    fontSize,
    fontWeight,
    fontFamily,
    textStyle,
    styledVariant({ variants: theme.textStyle }),
  ),
);

function Text({
  variant = 'bodyText',
  color = 'contentPrimary',
  ...rest
}: {
  variant?: TextVariantTypes;
  color?: string;
  [key: string]: any;
}) {
  return (
    <StyledText
      //@ts-ignore
      variant={variant}
      color={color}
      {...rest}>
      {rest.children}
    </StyledText>
  );
}

export default Text;
