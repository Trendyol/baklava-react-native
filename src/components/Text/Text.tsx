import { createText } from '@shopify/restyle';
import { Theme } from '../../theme';

const Text = createText<Theme>();

export default Text;
