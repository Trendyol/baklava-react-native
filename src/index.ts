export { ThemeProvider } from '@ergenekonyigit/restyle';
export { default as theme } from './theme';

export { default as Box } from './components/Box/Box';
export { default as Button } from './components/Button/Button';
export { default as Icon } from './components/Icon/Icon';
export { default as Text } from './components/Text/Text';
