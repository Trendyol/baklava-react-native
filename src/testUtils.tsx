import * as React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import theme from './theme';

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

export * from '@testing-library/react-native';

export { customRender as render };
