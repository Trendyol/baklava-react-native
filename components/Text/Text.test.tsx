import React from 'react';
import { theme } from '../../src';
import { render } from '../../src/testUtils';
import Text from '../Text/Text';

describe('Text', () => {
  test('should render text correctly', () => {
    // when
    const { toJSON } = render(<Text>testtesttest</Text>);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given text correctly', () => {
    // when
    const { getByTestId } = render(<Text testID="text">Test Text</Text>);
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.children).toBe('Test Text');
  });

  // Heading
  test('should render heading1 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading1">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('heading1');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[6]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[0]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render heading2 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading2">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('heading2');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[5]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  test('should render heading3 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading3">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('heading3');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[4]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle01
  test('should render subtitle01Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle01Regular');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[3]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle01Medium');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[3]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle01Semibold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[3]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[3]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle01Bold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[3]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[4]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle02
  test('should render subtitle02Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle02Regular');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[2]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle02Medium');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[2]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle02Semibold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[2]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[3]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle02Bold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[2]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[4]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle03
  test('should render subtitle03Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle03Regular');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle03Medium');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle03Semibold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[3]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle03Bold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[4]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle04
  test('should render subtitle04Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle04Regular');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle04Medium');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle04Semibold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[3]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('subtitle04Bold');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[4]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Body
  test('should render bodyText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('bodyText');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyUnderline variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyUnderline">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('bodyUnderline');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyTextLink variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyTextLink">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('bodyTextLink');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyLongText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyLongText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('bodyLongText');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[1]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Caption
  test('should render captionText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('captionText');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionMedium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionMedium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('captionMedium');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionLongText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionLongText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('captionLongText');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[1]);
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionTextLink variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionTextLink">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.variant).toBe('captionTextLink');
    expect(textComponent.props.style[0].fontSize).toBe(theme.fontSizes[0]);
    expect(textComponent.props.style[0].fontWeight).toBe(theme.fontWeights[2]);
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  //
});
