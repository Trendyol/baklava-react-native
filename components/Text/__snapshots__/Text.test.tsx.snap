// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text correctly 1`] = `
<Text
  color="contentPrimary"
  style={
    Array [
      Object {
        "color": "#273142",
        "fontFamily": "Rubik-Regular",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 16,
      },
    ]
  }
  variant="bodyText"
>
  testtesttest
</Text>
`;
